### PR TITLE
feat(ga4): enabled ga4 analytics on website and docs

### PR DESCRIPTION
**Pull request by Abhinandan-Purkait**
- Adds the gtag enabling script in website head.
- Integrates the docusaurus google tag plugin in docs.
- Integrates the new google tag manager to website on the react-gtm module.

### CODE REVIEW

File: docs/docusaurus.config.js [hunk 1]
```diff
@@ -43,7 +39,11 @@ module.exports = {
     },
     prism: {
       theme: Object.assign(require('prism-react-renderer/themes/github'), prismCustomColors)
-    }
+    },
+    gtag: {
+      trackingID: 'G-BNB79447GR',
```
Comment (tiagolobocastro) [inline docs/docusaurus.config.js:44]: Why is the tracing ID different now? Bear in mind I've got no idea how this stuff works

**Review by tiagolobocastro — APPROVED** [review]
I can't really provide any meaningful review here, other than looks reasonable

File: docs/docusaurus.config.js (hunk 1 above)
Comment (Abhinandan-Purkait) [reply docs/docusaurus.config.js:44]: Earlier it was not a GA4 id, it was an UA id. UA is deprecated now and so its all the docusaurus modules around it. Thus now we have to setup the GA4 analytics.

**Review by edrob999 — APPROVED** [review]
:thumbsup:

File: docs/docusaurus.config.js [hunk 2]
```diff
@@ -169,6 +169,13 @@ module.exports = {
         include: ["**/*.md", "**/*.mdx"], // Extensions to include.
       },
     ],
+    [
```
Comment (avishnu) [inline docs/docusaurus.config.js:172]: this wasn't there earlier?

File: website/src/index.tsx [hunk 3]
```diff
@@ -7,7 +7,7 @@ import './i18n';
 import Loader from './components/Loader';
 
 const tagManagerArgs = {
-  gtmId: 'GTM-KD8TCG4',
+  gtmId: 'GTM-WKVV7F3N',
```
Comment (avishnu) [inline website/src/index.tsx:10]: Shouldn't we use same id for both website and docs?

Comment (avishnu) [reply website/src/index.tsx:10]: Or is this a different id?

File: docs/docusaurus.config.js (hunk 2 above)
Comment (Abhinandan-Purkait) [reply docs/docusaurus.config.js:172]: We used to use the docusaurs-preset-classic's inbuilt UA integration earlier. But now we are using a standalone docusaurs plugin named plugin-google-gtag for the GA4.

File: website/src/index.tsx (hunk 3 above)
Comment (Abhinandan-Purkait) [reply website/src/index.tsx:10]: This is a gtagmanager container ID. Its a different entity that we can use for debugging and custom tagging.

File: docs/docusaurus.config.js (hunk 2 above)
Comment (Abhinandan-Purkait) [reply docs/docusaurus.config.js:172]: That's the reason I have added a new preset here.

**Pull request merged**